roles/icpc_host_backup/tasks/main.yml

```
@@ -2,21 +2,20 @@
 
 - name: Ensure packages that are needed for backup hosts are installed
   ansible.builtin.package:
```

```suggestion
  ansible.builtin.apt:
```
otherwise it doesn't work I think

---

I was following an example here: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_module.html

---

Yes, `package` is the suboptimal choice here as it needs the loop. Both the `apt` and `yum` modules can do this in 1 go.

---

you really think
```
ansible.builtin.apt:
    pkg:
    - foo
    - foo-tools
```

is handled different than:
```
ansible.builtin.package:
   name: 
       - foo
       - foo-tools
```

?  or that the docs are wrong?

---

i mean i replaced the loop with the name list.

---

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_module.html#parameter-name

name can be a `list of strings`

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_module.html#parameter-name
name can only be a string (so you need the loop, which leads to a lot of dependency checking)

So I don't think the docs are wrong here, I just think they are hard to read as between 2 modules which basically do the same thing they work completely different.

---

by docs i meant the example where they had name as a list of strings?

---

I still think they had an error in the package name declaration.   But i have switched to apt.